lib/honeydew/node_monitor.ex

```diff
@@ -4,6 +4,13 @@ defmodule Honeydew.NodeMonitor do
 
   @interval 1_000 # ms
 
+  def child_spec(opts) do
+    %{
+      id: __MODULE__,
```

Comment:
unless i'm mistaken, this is going to register the first node monitor process with the module name, preventing us from starting up other node monitors.

Reply:
You have to pass an `id` here. What would be appropriate if we wanted it to differ each time? Perhaps `{:node_monitor, :erlang.unique_integer()}`?

Reply:
sorry, i commented on the PR rather than directly here.

node monitors (and various other processes in the project) are started with a `:simple_one_for_one`, which is now `DynamicSupervisor`, they don't require an id, since the children represent identical functionality (they only differ by init arguments).

check this out:

```
defmodule NodeMonitor do
  use GenServer

  def start_link(node) do
    GenServer.start_link(__MODULE__, node)
  end

  def init(node) do
    IO.puts "i'm going to monitor #{node}!"
    {:ok, node}
  end
end

defmodule NodeMonitorSupervisor do
  use DynamicSupervisor

  def start_link() do
    DynamicSupervisor.start_link(__MODULE__, [])
  end

  def start_child(supervisor, node) do
    spec = {NodeMonitor, node}
    DynamicSupervisor.start_child(supervisor, spec)
  end

  def init(_) do
    DynamicSupervisor.init(strategy: :one_for_one)
  end
end

{:ok, supervisor} = NodeMonitorSupervisor.start_link()
NodeMonitorSupervisor.start_child(supervisor, :abc@computer)
```

Reply:
My understanding is that `use GenServer` will define a `child_spec/1` function whose returning map has an `id` of `__MODULE__`. DynamicSupervisor does appear to still require this `id` in the child spec (according to https://hexdocs.pm/elixir/DynamicSupervisor.html#module-migrating-from-supervisor-s-simple_one_for_one)

Perhaps I'm misunderstanding?

Reply:
i.e. it looks to me like the current implementation is doing `id: __MODULE__` under the hood anyway

Reply:
sorry, i meant to say that they don't require a dynamic id, since the id is that of the child specification, not that of the child process.

Reply:
Ok. I've read through some of the dynamic supervisor source code.. basically after converting the simple_one_to_one's to dynamic supervisors, it doesn't matter if we specify `id` in the child_spec, because it's ignored.

In the case of NodeMonitor, we can remove the `child_spec`, but for `Worker` it has to remain due to the way `extra_arguments` and the default child_spec for GenServer interact. 

(If you have extra_arguments of [1, 2, 3] and the default GenServer child_spec, then it tries calling the module with arguments `1, 2, 3, []`)

Reply:
https://github.com/elixir-lang/elixir/blob/v1.6/lib/elixir/lib/gen_server.ex#L583